app/routes.py

```diff
@@ -1,2 +1,188 @@
+
+        for task in tasks:
+            task_response.append(task.to_dict())
+        return jsonify(task_response), 200
+
+@tasks_bp.route("/<task_id>", methods= ["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    
+
+    if task is None:
+        return make_response(f"Task {task_id} not found"), 404
+        
+    if request.method == "GET":
+        return jsonify({"task": task.to_dict()}), 200
+        
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body:
+            return make_response("invalid request"), 400
+
+        task.title=request_body["title"]
+        task.description=request_body["description"]
+        
+
+        db.session.commit()
+        
+        return jsonify({"task": task.to_dict()}),200
+        
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+    
+        return ({'details': f'Task {task_id} "{task.title}" successfully deleted'}), 200
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    today = date.today()
+    if task is None:
+        return make_response("", 404)
+    else:
+        task.completed_at = today
+        db.session.commit()
+        PATH = 'https://slack.com/api/chat.postMessage'
+        params = {"token": os.environ.get("SLACK_TOKEN"),
+                    "channel":"task-notifications",
+                    "text":f"Someone just completed the task {task.title}"
+                    }
+        requests.post(PATH,data=params)
+        return jsonify({"task":task.to_dict()}), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    today = date.today()
+    if task is None:
+        return make_response("", 404)
+    else:
+        task.completed_at = None
+        db.session.commit()
+        
+        return jsonify({"task":task.to_dict()}), 200
+        
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body=request.get_json()
+    if "title" not in request_body:
+            return jsonify({"details": "Invalid data"}), 400 
+    new_goal=Goal(
+        title = request_body["title"]
+    )
+    db.session.add(new_goal)
+    db.session.commit()
+    
+    
+    return jsonify({"goal": {"id": new_goal.goal_id, "title": new_goal.title}}), 201
+
+
+@goals_bp.route("", methods=["GET"])
+def get_goals():
+    request_body=request.get_json()
+    
+    goals = Goal.query.all()
+    goal_response=[]
+    for goal in goals:
+        goal_response.append(goal.to_dict())
+        if not goal:
+            return make_response("", 404)
+    
+    return jsonify(goal_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal_id=int(goal_id)
```

Comment:
Make sure to do input validation here - if the user passed in an invalid `goal_id` this code would crash.